lib/jekyll/frontmatter_defaults.rb

```diff
@@ -97,21 +97,26 @@ def applies?(scope, path, type)
       applies_path?(scope, path) && applies_type?(scope, type)
     end
 
+    # rubocop:disable Metrics/AbcSize
     def applies_path?(scope, path)
       return true if !scope.key?("path") || scope["path"].empty?
 
       sanitized_path = Pathname.new(sanitize_path(path))
-
-      site_path = Pathname.new(@site.source)
+      site_path      = Pathname.new(@site.source)
       rel_scope_path = Pathname.new(scope["path"])
       abs_scope_path = File.join(@site.source, rel_scope_path)
-      Dir.glob(abs_scope_path).each do |scope_path|
-        scope_path = Pathname.new(scope_path).relative_path_from site_path
-        return true if path_is_subpath?(sanitized_path, scope_path)
+
+      if scope["path"].to_s.include?("*")
+        Dir.glob(abs_scope_path).each do |scope_path|
+          scope_path = Pathname.new(scope_path).relative_path_from site_path
+          Jekyll.logger.debug "Globbed Scope Path:", scope_path
+          return true if path_is_subpath?(sanitized_path, scope_path)
+        end
       end
 
       path_is_subpath?(sanitized_path, rel_scope_path)
```

Comment:
Could we make this an `else` for the previous block (`if scope["path"].to_s.include?("*")`)?
Multiple exits can get confusing.

Reply:
We can't make this an `else` branch because if `diff-ln#113` evaluates to `false` even at the end of looping the globbed array, `diff-ln#117` acts as a fallback..

That said, it does not make sense to me why that fallback's needed..